Firestore/core/src/core/firestore_client.cc

```diff
@@ -512,6 +514,47 @@ void FirestoreClient::RemoveSnapshotsInSyncListener(
   });
 }
 
+void FirestoreClient::LoadBundle(
+    std::unique_ptr<util::ByteStream> bundle_data,
+    std::shared_ptr<api::LoadBundleTask> result_task) {
+  VerifyNotTerminated();
+
+  bundle::BundleSerializer bundle_serializer(
+      remote::Serializer(database_info_.database_id()));
+  auto data = absl::ShareUniquePtr(std::move(bundle_data));
+  auto reader = std::make_shared<bundle::BundleReader>(
+      std::move(bundle_serializer), std::move(data));
+  worker_queue_->Enqueue([this, reader, result_task] {
+    sync_engine_->LoadBundle(std::move(reader), std::move(result_task));
+  });
+}
+
+void FirestoreClient::GetNamedQuery(const std::string& name,
+                                    api::QueryCallback callback) {
+  VerifyNotTerminated();
+
+  // Dispatch the result back onto the user dispatch queue.
+  auto async_callback =
+      [this, callback](absl::optional<bundle::NamedQuery> named_query) {
+        if (callback) {
+          if (named_query.has_value()) {
+            const Target& target = named_query.value().bundled_query().target();
+            Query query(target.path(), target.collection_group(),
+                        target.filters(), target.order_bys(), target.limit(),
+                        named_query.value().bundled_query().limit_type(),
+                        target.start_at(), target.end_at());
+            user_executor_->Execute([=] { callback(std::move(query)); });
```

Comment:
Nit: for lambdas that escape the current scope, it is recommended to list the captured variables explicitly (here and below).